### PR TITLE
Recent activity fetch in UI

### PR DESCRIPTION
**Pull request by fefrain**
This pull request has to be considered as a Work In Progress still.
The ActivityLogMessageParser has been splitted in several files (one for each module) and every activity log type needs to be addressed still.

### CODE REVIEW

File: server/modules/nodeDef/manager/nodeDefManager.js [hunk 1]
```diff
@@ -15,14 +15,12 @@ const ActivityLogRepository = require('@server/modules/activityLog/repository/ac
...
-  await client.tx(async t => {
-    const [nodeDef] = await Promise.all([
+  await client.tx(async t => (await Promise.all([
       NodeDefRepository.insertNodeDef(surveyId, nodeDefParam, t),
       markSurveyDraft(surveyId, t),
       ActivityLogRepository.insert(user, surveyId, ActivityLog.type.nodeDefCreate, nodeDefParam, system, t)
-    ])
-    return nodeDef
-  })
+    ]))[0]
```
Comment (minotogna) [inline server/modules/nodeDef/manager/nodeDefManager.js]: it was more readable before

File: server/modules/nodeDef/manager/nodeDefManager.js [hunk 2]
```diff
@@ -88,14 +86,16 @@ const updateNodeDefProps = async (user, surveyId, nodeDefUuid, props, propsAdvan
...
       : []
 
+    const nodeDef = await NodeDefRepository.updateNodeDefProps(surveyId, nodeDefUuid, props, propsAdvanced, t)
+
     const logContent = {
       uuid: nodeDefUuid,
       ...(R.isEmpty(props) ? {} : { props }),
-      ...(R.isEmpty(propsAdvanced) ? {} : { propsAdvanced })
+      ...(R.isEmpty(propsAdvanced) ? {} : { propsAdvanced }),
+      keys: Object.keys({ ...props, ...propsAdvanced }),
```
Comment (minotogna) [inline server/modules/nodeDef/manager/nodeDefManager.js]: why are you storing the keys? aren't them all keys except uuid?

File: test/it/006activityLogTest.js [hunk 3]
```diff
@@ -27,7 +27,7 @@ describe('Activity Log Test', async () => {
     const survey = await SurveyManager.createSurvey(getContextUser(), surveyParam)
     const surveyId = Survey.getId(survey)
 
-    const surveyCreateLogs = await ActivityLogRepository.fetch(surveyId, ActivityLog.type.surveyCreate)
+    const surveyCreateLogs = await ActivityLogRepository.fetch(surveyId, true, ActivityLog.type.surveyCreate)
```
Comment (minotogna) [inline test/it/006activityLogTest.js]: the second argument of ```ActivityLogRepository.fetch`` is activityTypes. i don't think it's a boolean. let's start adding assert to check the arguments?

File: server/modules/nodeDef/manager/nodeDefManager.js (hunk 1 above)
Comment (fefrain) [reply server/modules/nodeDef/manager/nodeDefManager.js]: right, I'll revert it to the previous version.

File: server/modules/nodeDef/manager/nodeDefManager.js (hunk 2 above)
Comment (fefrain) [reply server/modules/nodeDef/manager/nodeDefManager.js]: right, I can extract the keys later on in the MessageParser

File: test/it/006activityLogTest.js (hunk 3 above)
Comment (fefrain) [reply test/it/006activityLogTest.js]: right, I'll check it

Comment (eliask) [inline test/it/006activityLogTest.js]: I'm confused about what the second parameter does. Isn't fetch() typed like this?
```
export const fetch = async (surveyId, activityTypes = null, offset = 0, limit = 30, client = db) => ...
```

Comment (fefrain) [reply test/it/006activityLogTest.js]: the second parameter should be an array, right

File: server/modules/nodeDef/manager/nodeDefManager.js [hunk 4]
```diff
@@ -88,14 +88,15 @@ const updateNodeDefProps = async (user, surveyId, nodeDefUuid, props, propsAdvan
       ? await _updateNodeDefOnCyclesUpdate(surveyId, nodeDefUuid, props[NodeDef.propKeys.cycles], t)
       : []
 
+    const nodeDef = await NodeDefRepository.updateNodeDefProps(surveyId, nodeDefUuid, props, propsAdvanced, t)
```
Comment (minotogna) [inline server/modules/nodeDef/manager/nodeDefManager.js]: any reason to have moved this outside the Promise.all?

File: server/modules/activityLog/repository/activityLogRepository.js [hunk 5]
```diff
@@ -2,11 +2,12 @@ import camelize from 'camelize'
...
-import * as DbUtils from '@server/db/dbUtils'
 
 import * as ActivityLog from '@common/activityLog/activityLog'
 
+import * as db from '@server/db/db'
+import * as DbUtils from '@server/db/dbUtils'
+
 //===== CREATE
 export const insert = async (user, surveyId, type, content, system, client) =>
   await client.none(`
```
Comment (minotogna) [inline server/modules/activityLog/repository/activityLogRepository.js]: can you add assert checks here?

File: server/modules/activityLog/repository/activityLogRepository.js [hunk 6]
```diff
@@ -28,6 +29,7 @@ export const fetch = async (surveyId, activityTypes = null, offset = 0, limit =
       log AS
       (
         SELECT
+          a.id,
```
Comment (minotogna) [inline server/modules/activityLog/repository/activityLogRepository.js]: can you add assert checks to fetch method?

File: server/modules/nodeDef/manager/nodeDefManager.js (hunk 4 above)
Comment (fefrain) [reply server/modules/nodeDef/manager/nodeDefManager.js]: no... I'll fix it

File: server/modules/activityLog/repository/activityLogRepository.js (hunk 5 above)
Comment (fefrain) [reply server/modules/activityLog/repository/activityLogRepository.js]: ActivityLogRepository is used all over the application, I wouldn't do any check there.

**Pull request merged**